x-pack/plugins/security_solution/scripts/endpoint/sentinelone_host/index.ts

```diff
@@ -0,0 +1,161 @@
+                          can also be from 'SNAPSHOT'.
+                          Examples: 8.6.0, 8.7.0-SNAPSHOT
+      --policy            Optional. The UUID of the agent policy that should be used to enroll
+                          the Elastic Agent with Kibana/ES (Default: uses existing (if found) or
+                          creates a new one)
+      --force             Optional. If true, then fleet-server will be started and connected to
+                          kibana even if one seems to already be configured.
+      --username          Optional. User name to be used for auth against elasticsearch and
+                          kibana (Default: elastic).
+      --password          Optional. Password associated with the username (Default: changeme)
+      --kibanaUrl         Optional. The url to Kibana (Default: http://127.0.0.1:5601)
+      --elasticUrl        Optional. The url to Elasticsearch (Default: http://127.0.0.1:9200)
+`,
+    },
+  });
+};
+
+const runCli: RunFn = async ({ log, flags }) => {
+  const username = flags.username as string;
+  const password = flags.password as string;
+  const kibanaUrl = flags.kibanaUrl as string;
+  const elasticUrl = flags.elasticUrl as string;
+  const version = flags.version as string;
+  const s1Url = flags.s1Url as string;
+  const s1ApiToken = flags.s1ApiToken as string;
+  const policy = flags.policy as string;
+  const force = flags.force as boolean;
+
+  createToolingLogger.defaultLogLevel = flags.verbose
+    ? 'verbose'
+    : flags.debug
+    ? 'debug'
+    : flags.silent
+    ? 'silent'
+    : flags.quiet
+    ? 'error'
+    : 'info';
+
+  const getRequiredArgMessage = (argName: string) => `${argName} argument is required`;
+
+  ok(s1Url, getRequiredArgMessage('s1Url'));
+  ok(s1ApiToken, getRequiredArgMessage('s1ApiToken'));
+
+  const vmName =
+    (flags.vmName as string) ||
+    `${userInfo().username.toLowerCase().replaceAll('.', '-')}-sentinelone-${Math.random()
+      .toString()
+      .substring(2, 6)}`;
+  const s1Client = new S1Client({ url: s1Url, apiToken: s1ApiToken, log });
+  const { kbnClient } = await createRuntimeServices({
+    kibanaUrl,
+    elasticsearchUrl: elasticUrl,
+    username,
+    password,
+    log,
+  });
+
+  const hostVm = await createVm({
+    type: 'multipass',
+    name: vmName,
+    log,
+    memory: '2G',
+    disk: '10G',
+  });
+
+  const s1Info = await installSentinelOneAgent({
+    hostVm,
+    log,
+    s1Client,
+  });
+
+  if (force || !(await isFleetServerRunning(kbnClient))) {
+    await startFleetServer({ kbnClient, logger: log, version, force });
+  }
+
+  const agentPolicyId = policy || (await getOrCreateDefaultAgentPolicy({ kbnClient, log })).id;
+
+  await addSentinelOneIntegrationToAgentPolicy({
+    kbnClient,
+    log,
+    agentPolicyId,
+    consoleUrl: s1Url,
+    apiToken: s1ApiToken,
+  });
+
+  await enrollHostVmWithFleet({
+    hostVm,
+    kbnClient,
+    log,
+    agentPolicyId,
```

Comment:
I don't think we want to install elastic-agent on the S1 Host, the reason for that is that the users may not have 2 agents running on the same host and also for dev purposes, it's going to be better if we don't see this agent in Fleet, so we will not fail into relying/using Fleet data to see if it's S1 agent or not

Reply:
Thanks for the feedback. I guess I really am not sure how the data gets into Elasticsearch then. I thought we had to install the SentinelOne integration into the host - thus the Elastic agent install here

Reply:
FYI: I removed this option from the script. It will now do:

1. creates VM and installs SentinelOne Agent on it
2. Creates a Fleet Agent policy and adds the SentinelOne integration to it

Thanks